### PR TITLE
Fix data query in dashboard mode when enable around map filter

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
* Fix data render in dashboard mode when enable around map filter
* Refactor layer render functions

### Issue
Closes https://github.com/opensearch-project/dashboards-maps/issues/341

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/model/layerRenderController.ts [hunk 1]
```diff
@@ -136,10 +141,44 @@ export const handleDataLayerRender = (
...
+export const handleBaseLayerRender = (
   layer: MapLayerSpecification,
   maplibreRef: MaplibreRef,
   beforeLayerId: string | undefined
-) => {
+): void => {
   layersFunctionMap[layer.type].render(maplibreRef, layer, beforeLayerId);
 };
+
+export const handleAllDataLayersRender = (
```
Comment (VijayanB) [inline public/model/layerRenderController.ts:152]: nit
```suggestion
export const renderDataLayers = (
```

**Review by VijayanB — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,7 +29,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 * fix: fixed filters not reset when index pattern changed ([#234](https://github.com/opensearch-project/dashboards-maps/pull/234))
 * Add custom layer visibility config to render ([#297](https://github.com/opensearch-project/dashboards-maps/pull/297))
 * Fix color picker component issue ([#305](https://github.com/opensearch-project/dashboards-maps/pull/305))
-* fix: layer filter setting been reset unexpectedly ([#327](https://github.com/opensearch-project/dashboards-maps/commits/2.x))
+* fix: layer filter setting been reset unexpectedly ([#327](https://github.com/opensearch-project/dashboards-maps/pull/327))
+* Fix data query in dashboard mode when enable around map filter ([#339](https://github.com/opensearch-project/dashboards-maps/pull/339))
```
Comment (VijayanB) [inline CHANGELOG.md:33]: nit: shall we add empty line after 33?

Comment (junqiu-lei) [reply CHANGELOG.md:33]: I found each paragraph in the existing doc don't have empty line, and it doesn’t effect the markdown view. https://github.com/opensearch-project/dashboards-maps/blob/1e3f3b386fee8a9862f9cec1e5fa9965df57a9f8/CHANGELOG.md

Comment (junqiu-lei) [reply CHANGELOG.md:36]: Added line for each paragraph

Comment (VijayanB) [reply CHANGELOG.md:36]: Thanks, the advantage will be diff will not show previous line as part of change. In this case 35 is part of your diff but going forward only the added line will show up in diff.

**Review by VijayanB — APPROVED** [review]

**Review by heemin32 — APPROVED** [review]

**Pull request merged**